azure/defaults_test.go

```diff
@@ -235,3 +240,51 @@ func TestGetDefaultUbuntuImage(t *testing.T) {
 		})
 	}
 }
+
+func TestMSCorrelationIDSendDecorator(t *testing.T) {
```

Comment:
@CecileRobertMichon @devigned would you prefer that I include a test here for `SetAutoRestClientDefaults` rather than just `msCorrelationIDSendDecorator`. I didn't want to blow up the scope too much, but I'm happy to go a bit bigger in tests here if you'd prefer.

Reply:
this is fine with me, ideally we would add both but for this specific functionality a specific test is good enough